packages/components/fields/radio-field/src/radio-field.tsx

```diff
@@ -0,0 +1,229 @@
+
+const sequentialId = createSequentialId('radio-field-');
+
+const hasErrors = (errors?: TFieldErrors) =>
+  errors && Object.values(errors).some(Boolean);
+
+type TRadioFieldProps = {
+  // RadioField
+  /**
+   * Used as HTML id property. An id is auto-generated when it is not specified.
+   */
+  id?: string;
+  /**
+   * Horizontal size limit of the input fields.
+   */
+  horizontalConstraint?:
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+  /**
+   * A map of errors. Error messages for known errors are rendered automatically. Unknown errors will be forwarded to `renderError`.
+   */
+  errors?: TFieldErrors;
+  /**
+   * Called with custom errors.
+   * <br/>
+   * This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.
+   */
+  renderError?: TErrorRenderer;
+  /**
+   * Indicates if the value is required. Shows an the "required asterisk" if so.
+   */
+  isRequired?: boolean;
+  /**
+   * Indicates whether the field was touched. Errors will only be shown when the field was touched.
+   */
+  touched?: boolean;
+
+  // RadioInput
+  /**
+   * Used as HTML name of the input component.
+   */
+  name?: string;
+  /**
+   * Value of the input component.
+   */
+  value: string;
+  /**
+   * Called with an event containing the new value. Required when input is not read only. Parent should pass it back as value.
+   */
+  onChange?: ChangeEventHandler;
+  /**
+   * Called when input is blurred
+   */
+  onBlur?: FocusEventHandler;
+  /**
+   * Called when input is focused
+   */
+  onFocus?: FocusEventHandler;
+  /**
+   * Indicates that the input cannot be modified (e.g not authorized, or changes currently saving).
+   */
+  isDisabled?: boolean;
+  /**
+   * Indicates that the field is displaying read-only content
+   */
+  isReadOnly?: boolean;
+  /**
+   * Rendering direction of the radio `RadioInput.Option`s
+   */
+  direction?: 'stack' | 'inline';
+  /**
+   * Passes props of the `Spacings.Stack` or `Spacings.Inline`, dependeing on the chosen direction
+   */
+  directionProps?: object;
+  /**
+   * At least one `RadioInput.Option` component or another node (mixed children are allowed)
+   */
+  children: ReactNode;
+
+  // LabelField
+  /**
+   * Title of the label
+   */
+  title: string | ReactNode;
+  /**
+   * Hint for the label. Provides a supplementary but important information regarding the behaviour of the input (e.g warn about uniqueness of a field, when it can only be set once), whereas description can describe it in more depth.
+   * <br />
+   * Can also receive a `hintIcon`.
+   */
+  hint?: string | ReactNode;
+  /**
+   * Provides a description for the title.
+   */
+  description?: string | ReactNode;
+  /**
+   * Function called when info button is pressed.
+   * <br />
+   * Info button will only be visible when this prop is passed.
+   */
+  onInfoButtonClick?: () => void;
```

Comment:
Since this is forwarded to the `FieldLabel` component, maybe we should type the handler like it is over there:
```
onInfoButtonClick?: (
    event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
  ) => void;
```
or something like:
```
onInfoButtonClick?: (
    event: MouseEventHandler | KeyboardEventHandler
  ) => void;
```

What do you think?

Reply:
You're right. It's been updated. 
Thanks.